controllers/rollingupgrade_controller.go

```diff
@@ -177,3 +189,24 @@ func (r *RollingUpgradeReconciler) UpdateStatus(rollingUpgrade *v1alpha1.Rolling
 		r.Info("failed to update status", "message", err.Error(), "name", rollingUpgrade.NamespacedName())
 	}
 }
+
+func (r *RollingUpgradeReconciler) nodeReconciler(obj client.Object) []ctrl.Request {
```

Comment:
We should add handling of nodes getting deleted in case we ever need to check if a node is gone.
You can do this potentially by checking if `!ObjectMeta.DeletionTimestamp.IsZero()` which would indicate a deletion event.
If that is the case you can remove it from the sync map